### PR TITLE
[CROSSDATA-19] [Elasticsearch] Basic Select *

### PR DESCRIPTION
**Pull request by luismcl**
First commit: Support for Basic Select * From Table.

### CODE REVIEW

File: elastic-search/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.stratio.crossdata</groupId>
+        <artifactId>parent</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+    <packaging>jar</packaging>
+    <groupId>com.stratio.crossdata.connector</groupId>
+    <artifactId>elastic-search</artifactId>
```
Comment (darroyocazorla) [inline elastic-search/pom.xml]: I think elasticsearch would be easier

File: elastic-search/pom.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    <name>elastic-search</name>
+    <description>Crossdata Elastic-Search Connector</description>
+    <url>http://stratio.github.io/crossdata/</url>
+
+    <licenses>
```
Comment (darroyocazorla) [inline elastic-search/pom.xml]: the license is inherited from stratio-parent

File: elastic-search/pom.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        <license>
+            <name>Apache License, Version 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0</url>
+        </license>
+    </licenses>
+
+    <properties>
```
Comment (darroyocazorla) [inline elastic-search/pom.xml]: properties are interhited as well

File: elastic-search/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      val client = ElasticClient.remote(settings, host, port)
+
+      val resp: RichSearchResponse = client.execute(finalQuery).await
+
+      ElasticSearchRowConverter.asRows(schemaProvided.get, resp.hits)
+    }
+  }
+
+  def buildNativeQuery(requiredColumns: Array[Attribute], filters: Array[SourceFilter], query: SearchDefinition): SearchDefinition = {
+    val queryWithFilters = filtersToDBObject(filters, query)
```
Comment (darroyocazorla) [inline elastic-search/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala]: Remove DBObject reference

File: elastic-search/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    selectFields(requiredColumns, queryWithFilters)
+  }
+
+  private def filtersToDBObject(sFilters: Array[SourceFilter], query: SearchDefinition): SearchDefinition = {
+
+    val result = (query /: sFilters) { (prev, next) =>
```
Comment (darroyocazorla) [inline elastic-search/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala]: queryWithFilters instead of result?

**Comment by darroyocazorla** [comment]
LGTM

File: elasticsearch/pom.xml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        <dependency>
+            <groupId>com.sksamuel.elastic4s</groupId>
+            <artifactId>elastic4s-core_${scala.binary.version}</artifactId>
+            <version>${elastic4s.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>1.10.19</version>
```
Comment (miguel0afd) [inline elasticsearch/pom.xml]: The mockito version should be also a property

File: elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+ *
+ * @param logicalPlan [[LogicalPlan]]] to be executed
+ * @param parameters ElasticSearch Configuration Parameters
+ * @param schemaProvided Spark used defined schema
+ */
+class ElasticSearchQueryProcessor(val logicalPlan: LogicalPlan, val parameters: Map[String, String], val schemaProvided: Option[StructType] = None) extends Logging{
+
+  import ElasticSearchQueryProcessor._
+  /**
+   * Executes the [[LogicalPlan]]] and query the ElaticSearch database
```
Comment (miguel0afd) [inline elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala]: ElasticSearch, a "s" is missing

File: elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchRowConverter.scala [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    }.orNull
+  }
+
+  protected def enforceCorrectType(value: Any, desiredType: DataType): Any ={
+    if (value == null) {
+      null
+    } else {
+      desiredType match {
+        case StringType => value.toString
+        case _ if value == null || value == "" => null // guard the non string type
```
Comment (miguel0afd) [inline elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchRowConverter.scala]: I would move the logic of this line to the line 61 in order to group the cases where null must be returned according to the param value

File: pom.xml [hunk 9]
```diff
@@ -75,6 +75,7 @@
         <module>driver</module>
         <module>server</module>
         <module>dist</module>
+        <module>elasticsearch</module>
```
Comment (miguel0afd) [inline pom.xml]: I would set this module just after the other connectors

File: elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchRowConverter.scala (hunk 8 above)
Comment (luismcl) [inline elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchRowConverter.scala]: I'll fix on the next branch

**Pull request merged**